saml/response.go

```diff
@@ -87,15 +121,21 @@ func (sp *ServiceProvider) ParseResponse(
 	opt ...Option,
 ) (*core.Response, error) {
 	const op = "saml.(ServiceProvider).ParseResponse"
+	opts := getParseResponseOptions(opt...)
+
+	callValidateSignature := opts.validateResponseAndAssertionSignatures || opts.validateResponseSignature || opts.validateAssertionSignature
 	switch {
 	case sp == nil:
 		return nil, fmt.Errorf("%s: missing service provider %w", op, ErrInternal)
 	case samlResp == "":
 		return nil, fmt.Errorf("%s: missing saml response: %w", op, ErrInvalidParameter)
 	case requestID == "":
 		return nil, fmt.Errorf("%s: missing request ID: %w", op, ErrInvalidParameter)
+	case opts.skipSignatureValidation && callValidateSignature:
+		return nil, fmt.Errorf("%s: option `skip signature validation` cannot be true with any validate signature option : %w", op, ErrInvalidParameter)
+	case multipleSignatureOptionEnabled(opts.validateResponseAndAssertionSignatures, opts.validateResponseSignature, opts.validateAssertionSignature):
```

Comment:
I think I see discussion on these options already in the discussion, so I'll defer to `cap` people on the ergonomics/patterns, but if we need to check if the options are sensible, would it be better to reshape the options so that any pattern is valid? (for example, can we remove the "both" boolean and have `validateResponse` and `validateAssertion` both be true instead?

If there is a need/desire to have an explicit "both" option mutually exclusive with the other two, would it be worth turning it into a single field set to `const` values, e.g. 
```
const (
   NoSignatureValidation = iota
   ValidateAssertionSignature
   ValidateResponseSignature
   ValidateResponseAndAssertionSignature
)
```

I understand we wouldn't want to remove the `InsecureSkipValidation` functions, for compatibility reasons.

Reply:
@hcjulz  Do you have any recommendation for above w.r.t cap?

I like Kay point to remove the `both` option and ask user to set individual option instead.

Reply:
@mgaffney  just randomly tagging you as you are mentioned as the repo owner. 
Would you be able to assist in the review as Julian is out till end of the year, thanks

Let me know if you prefer/recommend someone else.

Reply:
Mike was not familiar with this piece of code..
tagging @jimlambrt  for review for Pr.

Reply:
I think, removing `ValidateResponseAndAssertionSignatures` makes for a better
dev experience.

Reply:
PR is updated now with only keeping two options